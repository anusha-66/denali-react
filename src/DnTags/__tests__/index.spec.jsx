/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import { DnTags, DnTagSizes, DnIcon } from '../../index'; // src/index.tsx

const frontIconComponent = <DnIcon icon="check" />;
const backIconComponent = <DnIcon icon="close" />;

describe('DnTags Tests', () => {
    it('should export DnTagSizes', () => {
        expect.assertions(1);

        expect(DnTagSizes).toStrictEqual({
            small: 'small',
            default: 'default',
        });
    });

    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnTags>
                <DnTags.Tag>Tag 1</DnTags.Tag>
                <DnTags.Tag>Tag 2</DnTags.Tag>
                <DnTags.Tag>Tag 3</DnTags.Tag>
                <DnTags.Tag>Tag 4</DnTags.Tag>
            </DnTags>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnTags nowrap className="dummy-tabs">
                <DnTags.Tag className="dummy-tab-item">Tag class name</DnTags.Tag>
                <DnTags.Tag isActive>Active Tag</DnTags.Tag>
                <DnTags.Tag isDisabled>Disabled Tag</DnTags.Tag>
                <DnTags.Tag isOutlined>Outlined Tag</DnTags.Tag>
                <DnTags.Tag size={DnTagSizes.small}>Small Tag</DnTags.Tag>
                <DnTags.Tag isActive isOutlined>
                    Active Outlined Tag
                </DnTags.Tag>
                <DnTags.Tag isActive isOutlined size={DnTagSizes.small}>
                    Active Outlined Small Tag
                </DnTags.Tag>
                <DnTags.Tag isDisabled isOutlined>
                    Disabled Outlined Tag
                </DnTags.Tag>
                <DnTags.Tag isDisabled isOutlined size={DnTagSizes.small}>
                    Disabled Outlined Small Tag
                </DnTags.Tag>
                <DnTags.Tag frontIconComponent={frontIconComponent}>Tag w/ Front Icon</DnTags.Tag>
                <DnTags.Tag backIconComponent={backIconComponent}>Tag w/ Back Icon</DnTags.Tag>
                <DnTags.Tag frontIconComponent={frontIconComponent} backIconComponent={backIconComponent}>
                    Tag w/ Front and Back Icon
                </DnTags.Tag>
            </DnTags>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
