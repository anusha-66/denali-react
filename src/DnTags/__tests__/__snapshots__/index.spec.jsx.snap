// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnTags Tests should render correctly with custom props 1`] = `
<div
  class="tags nowrap dummy-tabs"
>
  <span
    class="tag dummy-tab-item"
  >
    Tag class name
  </span>
  <span
    class="tag is-active"
  >
    Active Tag
  </span>
  <span
    class="tag is-disabled"
  >
    Disabled Tag
  </span>
  <span
    class="tag outlined"
  >
    Outlined Tag
  </span>
  <span
    class="tag is-small"
  >
    Small Tag
  </span>
  <span
    class="tag is-active outlined"
  >
    Active Outlined Tag
  </span>
  <span
    class="tag is-active outlined is-small"
  >
    Active Outlined Small Tag
  </span>
  <span
    class="tag is-disabled outlined"
  >
    Disabled Outlined Tag
  </span>
  <span
    class="tag is-disabled outlined is-small"
  >
    Disabled Outlined Small Tag
  </span>
  <span
    class="tag has-icon-front"
  >
    <i
      class="d-icon d-check is-default"
    />
    Tag w/ Front Icon
  </span>
  <span
    class="tag has-icon-back"
  >
    Tag w/ Back Icon
    <i
      class="d-icon d-close is-default"
    />
  </span>
  <span
    class="tag has-icon-front has-icon-back"
  >
    <i
      class="d-icon d-check is-default"
    />
    Tag w/ Front and Back Icon
    <i
      class="d-icon d-close is-default"
    />
  </span>
</div>
`;

exports[`DnTags Tests should render with default props 1`] = `
<div
  class="tags"
>
  <span
    class="tag"
  >
    Tag 1
  </span>
  <span
    class="tag"
  >
    Tag 2
  </span>
  <span
    class="tag"
  >
    Tag 3
  </span>
  <span
    class="tag"
  >
    Tag 4
  </span>
</div>
`;
